src/extension.ts

```diff
@@ -67,13 +67,20 @@ async function pickDevice(): Promise<void> {
         // user canceled
         return;
     }
-    ev3devBrowserProvider.setDevice(device);
-    try {
-        await device.connect();
-    }
-    catch (err) {
-        vscode.window.showErrorMessage(`Failed to connect to ${device.name}: ${err.message}`);
-    }
+    
+    await vscode.window.withProgress({
+        location: vscode.ProgressLocation.Window,
+        title: "Connecting..."
+    }, async progress => {
+        ev3devBrowserProvider.setDevice(device);
+        try {
+            await device.connect();

```

Comment:
I would still rather have `withProgress` around this line only.

Reply:
`setDevice` is also a potentially long-running function to my eyes. It ends a bunch of streams, closes connections, deallocates system resources, and invokes handlers.

Reply:
We are not awaiting it, so it runs synchronously. If takes more than 100ms to return, then we need to fix it. However, in practical usage, I have never noticed such a delay.

Reply:
Why would we want to do some work outside of the `withProgress` declaration? Even if we were confident that it would always be immediate, it is nonetheless logically part of the operation that is progressing. It doesn't make the code appreciably cleaner and arguably makes the flow more difficult to read.

Reply:
It still feels wrong to me, but this is too small of a thing to spend time arguing about it when no one will ever notice a difference either way.